src/Domain/Setting/test/SettingRepositoryStub.ts

```diff
@@ -0,0 +1,24 @@
+import { Setting } from '../Setting'
+import { SettingRepositoryInterface } from '../SettingRepositoryInterface'
+
+export class SettingRepostioryStub implements SettingRepositoryInterface {
+  constructor(private settings: Setting[]) {}
+
+  async findOneByNameAndUserUuid(name: string, userUuid: string): Promise<Setting | undefined> {
+    for (const setting of this.settings) {
+      if (setting.name === name && (await setting.user).uuid === userUuid) {
```

Comment:
@karolsojko @standarius out of curiosity, is it possible to read `setting.user_uuid` directly to avoid having to wait? Or is that non-idiomatic?

Reply:
Oh this is for tests, nevermind.

Reply:
as long as user is a promise it's not possible here
AFAIU `user_uuid` is the name of the column in the db that contains a reference to user uuid as a foreign key
the ORM framework then resolves that to a User object when fetching a Setting from the db

we could most likely make it not do that and just keep the user uuids -- then your use case would be supported

it is definitely simpler to use the uuids than promises -- this way you can serialize the settings to JSON without transforming, and don't have to deal with the awaiting

and you can always zip the collections by uuid if necessary

but you do lose some ORM magic and at this point it would probably require significant effort to make it work like this

Reply:
Not possible due to lazy loading - see: https://github.com/standardnotes/auth/pull/6#discussion_r604057931